app/src/main/java/org/example/ChessBoard.java

```diff
@@ -0,0 +1,109 @@
+package org.example;
+
+import org.example.ChessPieces.ChessPiece;
+import org.example.ChessPieces.ChessPieceGenerator;
+
+public class ChessBoard {
+    private ChessPiece[][] mBoard;
+
+    private BoardPosition mActivePiecePosition;
+    private boolean[][] mActivePieceLegalMoves;
+
+    ChessBoard(String[] boardLayout) {
+
+        ChessPieceGenerator generator = new ChessPieceGenerator();
+
+        // TODO: Check if it's useful to declare the size of 'mBoard' dynamically
+        // instead of a fixed 8x8 size
+        mBoard = new ChessPiece[boardLayout.length][boardLayout[0].length()];
+
+        for (int i = 0; i < mBoard.length; i++) {
+            for (int j = 0; j < mBoard[i].length; j++) {
+                char currPiece = boardLayout[i].charAt(j);
+
+                // TODO: flip 'i' to match GUI ordering:
+                mBoard[7 - i][j] = generator.fromString(currPiece);
+            }
+        }
+    }
+
+    char getPieceCharFromPos(BoardPosition pos) {
+        if (pos.isValid() && mBoard[pos.i][pos.j] != null) {
```

Comment:
`pos.isValid()` the whole point of BoardPosition was to represent only valid positions.

In other words, if BoardPosition exists, it's valid.

Reply:
Agreed. This was a redundant check. It also existed across a few other methods within `ChessBoard`, I updated them all.